llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -889,6 +889,23 @@ class YkIRWriter {
     InstIdx++;
   }
 
+  /// Serialise ptrtoint instruction.
+  void serialisePtrToIntInst(PtrToIntInst *I, FuncLowerCtxt &FLCtxt, unsigned BBIdx,
+                         unsigned &InstIdx) {
+    // opcode:
+    serialiseOpcode(OpCodeCast);
+    // cast_kind:
+    serialiseCastKind(CastKindZeroExt);
```

Comment:
Just reading the LLVM reference, `ptrtoint` has more semantics than we first thought:
https://llvm.org/docs/LangRef.html#ptrtoint-to-instruction

 - We should reject the vector variant (encode an unimplemented instruction). You can copy/adapt existing code from other opcodes that have vector variants.
 - `ptrtoint` can truncate to a smaller integer size, so zero extension isn't generally correct. If you don't want to handle those now, you can make an unimplemented instruction if the output type isn't pointer sized.

Reply:
(when rejecting certain versions of instructions, I've also been adding a case to `tests/ir_lowering/unsupported_variants.ll` to check they are properly rejected)

Reply:
My first instinct after your comment was to just add a `aot_ir::PtrToIntInst`. But reading the langref, it does say they either truncate/zeroextend/noop anyway. So we can probably do without such an instruction and just lower each case to different existing aot instructions.

Reply:
that would be fine by me.